### DISCUSSION TITLE
Can I haz WASM?

### DISCUSSION THREAD

**Discussion by larscwallin** [jhy/jsoup#2072] [Q&A]
Hi 🙂

Would it be possible to compile JSoup to WASM using a tool such as TeaVM you think?

Thanks for your excellent work!

**Comment by jhy** [comment]
It's an interesting idea. I'm not super deep on TeaVM but from a quick look it seems possible. jsoup doesn't have any dependencies other than the standard JDK lib so there are not too many moving parts.

Can you expand on the use case - how are you using jsoup, why do you want to run it in a browser (?), why not just use the browser's parser / DOM methods?

**Reply by larscwallin** [reply]
Hey thanks for the reply 🙂
It would give us a cross platform DOM parser. It could be run on iOS (JavaScriptCore) / Android (native code) / NodeJS. It could also be run in Web worker contexts to add multi threaded DOM parsing to Web apps.
I will try to see if I can get it to work and report back 👍
I might ask a question or two again 😉

**Reply by larscwallin** [reply]
PS. Today we use it today to parse HTML in our Android framework.

**Reply by itboy87** [reply]
@larscwallin  You can check out https://github.com/fleeksoft/ksoup. WebAssembly (wasm) support is already in progress, and it currently supports JVM, Android, iOS, Linux, JavaScript, and Windows.

**Reply by larscwallin** [reply]
Ooh, that's nice! Thanks 😃
